rig-core/src/providers/eternalai.rs

```diff
@@ -331,6 +337,68 @@ pub fn get_chain_id(key: &str) -> Option<&str> {
     None
 }
 
+pub async fn get_on_chain_system_prompt(
+    rpc_url: &str,
+    contract_addr: &str,
+    agent_id: c_uint,
+) -> Option<String> {
+    abigen!(
+        SystemPromptManagementContract,
+        r#"
+        [{"inputs": [{"internalType": "uint256", "name": "_agentId", "type": "uint256"}], "name": "getAgentSystemPrompt", "outputs": [{"internalType": "bytes[]", "name": "","type": "bytes[]"}], "stateMutability": "view", "type": "function"}]
+        "#
+    );
+    // Connect to an Ethereum node
+    let provider = Provider::<Http>::try_from(rpc_url).expect("Failed to parse url");
+    let client = Arc::new(provider);
+    let contract_address: Address = contract_addr.parse().expect("invalid contract address");
+    let contract = SystemPromptManagementContract::new(contract_address, client);
```

Comment:
Would be nice to store all those values in some sort of `OnChainClient` struct so we don't have to initialize them every time the function is called